src/lib/stepper/stepper-vertical.html

```diff
@@ -6,19 +6,28 @@
        [tabIndex]="_focusIndex == i ? 0 : -1"
        (click)="step.select()"
        (keydown)="_onKeydown($event)">
-    <div class="mat-stepper-index">
+    <div class="mat-stepper-index-new" *ngIf="!step.completed || selectedIndex == i">
       {{i + 1}}
     </div>
+    <div class="mat-stepper-index-interacted" *ngIf="step.completed && selectedIndex != i">
```

Comment:
Yeah, I meant an internal directive for the shared functionality. If they have a lot in common, would be easier to update together.